src/chainparams.cpp

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) 2009-2020 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "chainparams.h"
+
+#include "tinyformat.h"
+#include "util/strencodings.h"
+
+#include <assert.h>
+
+#include <boost/algorithm/string/classification.hpp>
+#include <boost/algorithm/string/split.hpp>
+
+/* TODO: Uncomment this when CBlock is moved from main.h
+static CBlock CreateGenesisBlock(const char* pszTimestamp, uint32_t nTime, uint32_t nNonce, uint32_t nBits, int32_t nVersion)
+{
+    CMutableTransaction txNew;
+    txNew.nVersion = 1;
+    txNew.nTime = nTime;
+    txNew.vin.resize(1);
+    txNew.vout.resize(1);
+    txNew.vin[0].scriptSig = CScript() << 0 << CBigNum(42) << vector<unsigned char>((const unsigned char*)pszTimestamp, (const unsigned char*)pszTimestamp + strlen(pszTimestamp));
+    txNew.vout[0].SetEmpty();
+
+    CBlock genesis;
+    genesis.nTime    = nTime;
+    genesis.nBits    = nBits;
+    genesis.nNonce   = nNonce;
+    genesis.nVersion = nVersion;
+    genesis.vtx.push_back(txNew);
+    genesis.hashPrevBlock.SetNull();
+    genesis.hashMerkleRoot = genesis.BuildMerkleTree();
+    return genesis;
+} */
+
+/**
+ * Build the genesis block. Note that the output of its generation
+ * transaction cannot be spent since it did not originally exist in the
+ * database.
+ *
+ * CBlock(hash=00000f762f698b5962aa81e38926c3a3f1f03e0b384850caed34cd9164b7f990, ver=1,
+ *        hashPrevBlock=0000000000000000000000000000000000000000000000000000000000000000,
+ *        hashMerkleRoot=0bd65ac9501e8079a38b5c6f558a99aea0c1bcff478b8b3023d09451948fe841, nTime=1413149999, nBits=1e0fffff, nNonce=1572771, vtx=1, vchBlockSig=)
+static CBlock CreateGenesisBlock(uint32_t nTime, uint32_t nNonce, uint32_t nBits, int32_t nVersion)
+{
+    const char* pszTimestamp = "10/11/14 Andrea Rossi Industrial Heat vindicated with LENR validation";
+    return CreateGenesisBlock(pszTimestamp, nTime, nNonce, nBits, nVersion);
+}  */
+
+/**
+ * Main network
+ */
+class CMainParams : public CChainParams {
+public:
+    CMainParams() {
+        strNetworkID = CBaseChainParams::MAIN;
+        consensus.ProtocolV2Height = 85400;
+        consensus.ResearchAgeHeight = 364501;
+        consensus.BlockV8Height = 1010000;
+        consensus.BlockV9Height = 1144000;
+        consensus.BlockV9TallyHeight = 1144120;
+        consensus.BlockV10Height = 1420000;
+        consensus.BlockV11Height = 2053000;
+        /**
+         * The message start string is designed to be unlikely to occur in normal data.
+         * The characters are rarely used upper ASCII, not valid as UTF-8, and produce
+         * a large 32-bit integer with any alignment.
+         */
+        pchMessageStart[0] = 0x70;
+        pchMessageStart[1] = 0x35;
+        pchMessageStart[2] = 0x22;
+        pchMessageStart[3] = 0x05;
+        nDefaultPort = 32749;
+        m_assumed_blockchain_size = 2;
```

Comment:
```suggestion
        m_assumed_blockchain_size = 4;
```
```
du -hs blk000* txleveldb/
2.0G    blk0001.dat
437M    blk0002.dat
1.2G    txleveldb/
```
We don't have a chainstate database yet, so the block flat files and index count together (and this gives a bit of wiggle room).